### PR TITLE
refactor website client to handle a membership object

### PR DESCRIPTION
**Pull request by ahsu1230**
@andrewconner @2is10 

change `getLibrary` (by user/slug or id) to handle a `membership` object inside the `library` object instead of separate sibling fields `membership`, `access`, `subscribed`, `listed`.

If a user does not have access to a library, there should be no `membership` object

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/libraries/library.js [hunk 1]
```diff
@@ -280,7 +280,7 @@ angular.module('kifi')
 
     $rootScope.$emit('libraryOnPage', library);
 
-    if (!libraryService.isLibraryMainOrSecret(library) && library.access !== 'none') {
+    if (!libraryService.isLibraryMainOrSecret(library) && library.membership) {
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/libraries/library.js:283]: This reads better. Nice bonus.

File: kifi-backend/modules/shoebox/angular/src/libraries/libraryHeader.js [hunk 2]
```diff
@@ -634,9 +634,14 @@ angular.module('kifi')
           }),
           $rootScope.$on('libraryJoined', function (e, libraryId) {
             var lib = scope.library;
-            if (lib && libraryId === lib.id && lib.access === 'none') {
+            if (lib && libraryId === lib.id && !lib.membership) {
+              (lib.invite || {}).actedOn = true;
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/libraries/libraryHeader.js]: Harmless, but this `.actedOn` line shouldn’t be added here. I removed it last week.

Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/libraries/libraryHeader.js]: oh I see. no wonder.

File: kifi-backend/modules/shoebox/angular/src/libraries/manageLibrary.js [hunk 3]
```diff
@@ -74,13 +74,13 @@ angular.module('kifi')
...
             visibility: scope.library.visibility,
-            listed: scope.library.listed,
+            listed: scope.library.membership.listed,
             color: colorNames[scope.library.color]
           }, true).then(function (resp) {
             libraryService.fetchLibraryInfos(true);
 
             var newLibrary = resp.data.library;
-            newLibrary.listed = resp.data.listed;
+            newLibrary.listed = resp.data.listed || (resp.data.library.membership && resp.data.library.membership.listed);
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/libraries/manageLibrary.js:83]: Odd that this `.listed` is still on a library object. Maybe we should rethink having the value in this dialog. Everything else there is a property of the library, not the owner/collaborator’s membership. Maybe the checkbox in this form should instead control the default or “maximum” value of `.listed` for all collaborators or even all members.

File: kifi-backend/modules/shoebox/angular/src/libraries/manageLibrary.tpl.html [hunk 4]
```diff
@@ -48,7 +48,7 @@ <h1 class="dialog-header">
               <div ng-switch-default class="manage-lib-visibility-description-content">
                 Anyone can see and follow this library
                 <div class="manage-lib-listed-visibility">
-                  <input type="checkbox" id="listed-visibility-checkbox" ng-model="library.listed">
+                  <input type="checkbox" id="listed-visibility-checkbox" ng-model="library.membership.listed">
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/libraries/manageLibrary.tpl.html:51]: In `openModifyLibrary` in userProfileLibraries.js, I see:

``` js
library.listed = data.listed;
```

It should now set `.membership.listed` instead, no?

Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/libraries/manageLibrary.tpl.html:51]: There’s another use of `lib.listed` in userProfileLibrariesList.tpl.html.

**Comment by 2is10** [comment]
Nice work. This is tricky.

File: kifi-backend/modules/shoebox/angular/src/libraries/manageLibrary.tpl.html (hunk 4 above)
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/angular/src/libraries/manageLibrary.tpl.html:51]: yup just edited both!

**Comment by ahsu1230** [comment]
oh yea, btw: with this `membership` will be in the `library` object and PR #15474 will change accordingly

**Pull request merged**